src/main/webapp/app/shared/util/firebase/firebase-utils.tsx

```diff
@@ -825,15 +849,15 @@ export const getTreatmentStats = (treatment?: Treatment) => {
     dxLevels: {} as { [dxLevel in DX_LEVELS]: number },
     pxLevels: {} as { [pxLevel in PX_LEVELS]: number },
   };
-  if (isTxLevelPresent(treatment.level)) {
-    stats.txLevels[treatment.level] = 1;
+  if (isTxLevelPresent(treatment?.level)) {
+    stats.txLevels[treatment?.level] = 1;
   }
   return stats;
 };
 
-export const getReviewInfo = (editor: string, action: string, updateTime?: string) => {
+export const getReviewInfo = (editor: string | undefined, action: string | undefined, updateTime?: string) => {
   const baseText = `${action} by ${editor}`;
-  let timeComponent;
+  let timeComponent: JSX.Element | undefined = undefined;
```

Comment:
This is somewhere I do expect the first two arguments to be defined, otherwise the caller of this function should not call it, or this util function should return a default. If you leave a comment here, I can come back and implement the first suggestion.

Reply:
I added `?? ''` to `const editor = reviewLevel?.reviewInfo.review.updatedBy ?? '';` in the `ReviewCollapsible` component and removed the `undefined` definitions in the arguments. Seems like everything's working now.